federator/src/lib/FederatorNFT.ts

```diff
@@ -0,0 +1,372 @@
+import { Logger } from 'log4js';
+import { Config } from '../../config/configts';
+import Web3 from 'web3';
+
+import web3 from 'web3';
+import fs from 'fs';
+import TransactionSender from './TransactionSender';
+import CustomError from './CustomError';
+import BridgeFactory from '../contracts/BridgeFactory';
+import FederationFactory from '../contracts/FederationFactory';
+import utils from './utils';
+
+const NFT_CONFIRMATION_BLOCKS = 5;
+export class FederatorNFT {
+  public logger: Logger;
+  public config: Config;
+  public mainWeb3: Web3;
+  public sideWeb3: Web3;
+  public sideFederationAddress: string = null;
+  public transactionSender: TransactionSender;
+  public bridgeFactory: BridgeFactory;
+  public federationFactory: FederationFactory;
+  public nftConfirmationsBlocks: number;
+  private federatorContract: import('../contracts/IFederationV2');
+
+  constructor(config: Config, logger: Logger, Web3 = web3) {
+    this.config = config;
+    this.logger = logger;
+    this.nftConfirmationsBlocks = config.nftConfirmations;
```

Comment:
In this cases it's easier a ternary if or when checking for null, empty or undefined just use ??
```suggestion
    this.nftConfirmationsBlocks = config.nftConfirmations ?? NFT_CONFIRMATION_BLOCKS;
```

Reply:
done at https://github.com/rsksmart/tokenbridge/pull/257/commits/960a77254356ad684efe45071b291b518e0266f8